packages/design-system/src/components/Badge/variants/BadgeDropdown.tsx

```diff
@@ -91,17 +87,18 @@ const BadgeDropdown = forwardRef((props: BadgeDropdownProps, ref: Ref<HTMLSpanEl
 
 	const { t } = useTranslation(I18N_DOMAIN_DESIGN_SYSTEM);
 
-	const [selectedValue, setSelectedValue] = useState(
-		selectedId ? value.find(v => v.id === selectedId) : value[0],
-	);
+	const selectedValue = value.find(v => v.id === selectedId) || value[0];
 
 	return (
 		<BadgePrimitive {...props} ref={ref}>
 			<Dropdown
 				aria-label={t('BADGE_ARIA_lABEL_SELECT_ITEM', 'Select item')}
-				items={value.map(mapBadgeItemToDropdownItem(setSelectedValue, onChange))}
+				items={value.map(mapBadgeItemToDropdownItem(onChange))}
+				data-testid={props['data-testid']}
```

Comment:
is there a reason no to add data-test here ?

Reply:
the Dropdown build both with this value, i can duplicate it but it depends if we want to have different values for test and testid or not

Reply:
now sending both values